lib/src/results.dart

```diff
@@ -207,3 +210,34 @@ extension RealmResultsInternal on RealmResults {
     return RealmResults<T>._(handle, realm);
   }
 }
+
+class RealmResultsIterator<T extends RealmObject> implements Iterator<T> {
+  final Iterable<T> _iterable;
+  final int _length;
+  int _index;
+  T? _current;
```

Comment:
[q] What happens if `T` is already nullable? I.e. if I have a `RealmResults<Person?>`, will this work?

Reply:
In Dart nullable types are a little bit different than .Net. Since generic type extends RealmObject (not RealmObject?), it is not possible to pass nullable type. So I think it is OK to be like that.

Reply:
Ugh... I think this will be a problem down the line as `RealmResults` should be able to contain `null` objects. This is not the case for regular queries like `realm.all<Foo>`, but when you add support for dictionaries, the keys and the values are represented as `RealmResults` and the values collection can contain `null` objects. But it doesn't have to be fixed in this PR - it's more of a deeper reaching design change for `RealmResults` in general.

Reply:
I created a task for that for the alpha, we need to support this

Reply:
Since @blagoev has created issue #163 I think we can resolve this discussion. @nirinchev, thank you for the helpful advice!